txp/Pos/Txp/Logic/Local.hs

```diff
@@ -171,15 +172,15 @@ txProcessTransactionNoLock itw@(txId, txAux) = reportTipMismatch $ runExceptT $
 txNormalize
     :: ( TxpLocalWorkMode ctx m
        , MonadSlots ctx m)
-    => m ()
-txNormalize = getCurrentSlot >>= \case
+    => MemPoolSnapshot -> m ()
+txNormalize memPoolSnapshot = getCurrentSlot >>= \case
     Nothing -> do
         tip <- GS.getTip
         -- Clear and update tip
         setTxpLocalData (mempty, def, mempty, tip, def)
     Just (siEpoch -> epoch) -> do
         utxoTip <- GS.getTip
-        localTxs <- getLocalTxs
+        let localTxs = getLocalTxs memPoolSnapshot
```

Comment:
See the `setTxpLocalData` above? We'll have to deal with updates alongside these reads, we can't do them fully independently or we'll get it wrong.